### ISSUE TITLE
Error: local variable 'config_trape' referenced before assignment

### ISSUE THREAD

**Issue by Pentest00111234** [jofpin/trape#215]
When i set all API , then this error has show . And then  loop the ngrok setup.
Please let me know , how to fixed it?

**Comment by Kali95739** [comment]
News flash the maker doesn't give a shit about this tool it's broke use beef xss

**Comment by aeroxy** [comment]
lol good to know @Kali95739

**Comment by ofer026** [comment]
@Pentest00111234 you are right! this error is driving me nuts and i don't know what to do

**Comment by Kuro-Arigami** [comment]
Hello my good friends here is the solution to the loop.
Steps:
1. Visit your trape folder.
2 Open the core folder and you'll find trape.py.
3. Open the trape.py file and do what I'll illustrate in step four.
4. Do the following open trape.py in the core folder and:

    find:
v = '{\n\t"ngrok_token" : "' + c_nGrokToken + '",\n\t"gmaps_api_key" : "' + c_gMapsToken + '",\n\t"gshortener_api_key" : "' + c_gOoglToken + '"\n\t"ipinfo_api_key" : "' + c_ipinfo + '",\n}' 

 replace with:
v = '{\n\t"ngrok_token" : "' + c_nGrokToken + '",\n\t"gmaps_api_key" : "' + c_gMapsToken + '",\n\t"gshortener_api_key" : "' + c_gOoglToken + '",\n\t"ipinfo_api_key" : "' + c_ipinfo + '"\n}'

It will solve the looping error after doing this find me and buy me lunch :) :) :)

**Comment by WWWWSS** [comment]
@Kuro-Arigami not working for me

**Comment by trash3d** [comment]
works for me
https://www.youtube.com/watch?v=bm5120HSD0Q

**Comment by sumitprajapat296** [comment]
how fix this error??

**Comment by LionOfIsrael** [comment]
> Hello my good friends here is the solution to the loop.
> Steps:
> 
> 1. Visit your trape folder.
>    2 Open the core folder and you'll find trape.py.
> 2. Open the trape.py file and do what I'll illustrate in step four.
> 3. Do the following open trape.py in the core folder and:
>    find:
>    v = '{\n\t"ngrok_token" : "' + c_nGrokToken + '",\n\t"gmaps_api_key" : "' + c_gMapsToken + '",\n\t"gshortener_api_key" : "' + c_gOoglToken + '"\n\t"ipinfo_api_key" : "' + c_ipinfo + '",\n}'
> 
> replace with:
> v = '{\n\t"ngrok_token" : "' + c_nGrokToken + '",\n\t"gmaps_api_key" : "' + c_gMapsToken + '",\n\t"gshortener_api_key" : "' + c_gOoglToken + '",\n\t"ipinfo_api_key" : "' + c_ipinfo + '"\n}'
> 
> It will solve the looping error after doing this find me and buy me lunch :) :) :)

This is the same exact code.

**Comment by roltsolen** [comment]
> Hello my good friends here is the solution to the loop.
> Steps:
> 
> 1. Visit your trape folder.
>    2 Open the core folder and you'll find trape.py.
> 2. Open the trape.py file and do what I'll illustrate in step four.
> 3. Do the following open trape.py in the core folder and:
>    find:
>    v = '{\n\t"ngrok_token" : "' + c_nGrokToken + '",\n\t"gmaps_api_key" : "' + c_gMapsToken + '",\n\t"gshortener_api_key" : "' + c_gOoglToken + '"\n\t"ipinfo_api_key" : "' + c_ipinfo + '",\n}'
> 
> replace with:
> v = '{\n\t"ngrok_token" : "' + c_nGrokToken + '",\n\t"gmaps_api_key" : "' + c_gMapsToken + '",\n\t"gshortener_api_key" : "' + c_gOoglToken + '",\n\t"ipinfo_api_key" : "' + c_ipinfo + '"\n}'
> 
> It will solve the looping error after doing this find me and buy me lunch :) :) :)

Sir you saved my life I got it fixed. I made an account just to thank you

**Comment by Kali95739** [comment]
Yeah it works now thanks but why is it going so slow it takes like 20 minutes to load the panel page and the hook page doesn't even load is anyone else having this problem?

**Comment by naimkowshik** [comment]
> Hello my good friends here is the solution to the loop.
> Steps:
> 
>     1. Visit your trape folder.
>        2 Open the core folder and you'll find trape.py.
> 
>     2. Open the trape.py file and do what I'll illustrate in step four.
> 
>     3. Do the following open trape.py in the core folder and:
>        find:
>        v = '{\n\t"ngrok_token" : "' + c_nGrokToken + '",\n\t"gmaps_api_key" : "' + c_gMapsToken + '",\n\t"gshortener_api_key" : "' + c_gOoglToken + '"\n\t"ipinfo_api_key" : "' + c_ipinfo + '",\n}'
> 
> 
> replace with:
> v = '{\n\t"ngrok_token" : "' + c_nGrokToken + '",\n\t"gmaps_api_key" : "' + c_gMapsToken + '",\n\t"gshortener_api_key" : "' + c_gOoglToken + '",\n\t"ipinfo_api_key" : "' + c_ipinfo + '"\n}'
> 
> It will solve the looping error after doing this find me and buy me lunch :) :) :)

**Thank You Bro solve this problem** 

https://prnt.sc/sy8d5s

**Comment by Orangeeye777** [comment]
[x] ERROR: local variable 'config_trape' referenced before assignment 

ISSUE RESOLVED :

Update the details properly while configuring the trape.config
i have use below site, create the account
1. ngrok - for ngrok_token
2. app.ipgeolocation.io - for gmaps_api_key
3. No need to update google url shorner
4. ipinfo.io - for ipinfo_api_key
Then hit ENTER. we are done...

**Comment by DanteMex** [comment]
correct API 
Google Search Console API
Google
The Search Console API provides access to Search Console data for your verified websites.
API Enabled
- Congratulations! Successful configuration, now enjoy Trape!
